tests/testAutopas/tests/utils/ArrayUtilsTest.cpp

```diff
@@ -40,3 +42,38 @@ TEST(ArrayUtilsTest, testto_string) {
   EXPECT_EQ("[1, 2, 3]", utils::ArrayUtils::to_string(bStringContainer));
   EXPECT_EQ("1x2x3", utils::ArrayUtils::to_string(bStringContainer, "x", {"", ""}));
 }
+
+/**
+ * Test utils::ArrayUtils::balanceVectors.
+ * Creates a 2D Vector, rebalances it and checks if the lengths are what we expect.
+ */
+TEST(ArrayUtilsTest, testBalanceVectors) {
+  constexpr size_t numVectors = 8;
+  // Setup some vectors of random length between 0 and 10
+  std::mt19937 g(42);
+  std::uniform_int_distribution dist(0, 10);
+  std::array<std::vector<size_t>, numVectors> vecvec{};
+  for (auto &v : vecvec) {
+    v.resize(dist(g));
+  }
+
+  // count the number of "inserted" elements
+  const auto numElements = [&]() {
+    size_t sum = 0;
+    for (const auto &vec : vecvec) {
+      sum += vec.size();
+    }
+    return sum;
+  }();
+  // calculate expectations
+  const auto expectedPerVec = numElements / vecvec.size();
+  const auto rest = numElements % vecvec.size();
```

Comment:
Similar to `ArrayUtils.h`, use a different name for `rest` that's clearer

Reply:
done 6685dbbb05334e01eb1001ad4b603bcceed1b05b